htdocs/core/class/commonobject.class.php

```diff
@@ -6144,7 +6145,13 @@ public function showInputField($val, $key, $value, $moreparam = '', $keysuffix =
 				$sql .= $sqlwhere;
 				//print $sql;
 
-				$sql .= ' ORDER BY '.implode(', ', $fields_label);
+				//fix #22571 : order by could be set
+				//remember 'sellist:TableName:LabelFieldName[:KeyFieldName[:KeyFieldParent[:Filter[:Sortfield]]]]',
+				if(isset($InfoFieldList[5]) && $InfoFieldList[5] != "") {
```

Comment:
For extrafield that must follow same rule than commonfield, the field 5 and 6 were already reserved.
Is it possible to add the sort as field at position 7 ?

Reply:
ho sorry for the delay, let me try it of course !

Reply:
@eldy updated